kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/DataSkippingUtils.java

```diff
@@ -260,4 +311,54 @@ private static Predicate reverseComparatorFilter(Predicate predicate) {
             getLeft(predicate)
         );
     }
+
+    /**
+     * Prunes the given schema according to the provided set of top-level columns to keep. The
+     * columns in {@code schema} are taken to be nested under {@code parentPath}.
+     *
+     * @param schema schema to prune
+     * @param columnsToKeep set of top-level columns to keep
+     * @param parentPath parent path of the fields in {@code schema} relative to the top-level
+     *                   schema
+     */
+    private static StructType pruneSchema(
```

Comment:
I reconsidered changing `Column` to take a List instead of Array here. But I think this doesn't make much of a difference considering we will need to create _new_ lists every time we create a column anyways (or each time we recurse into a nested field).